feature/home/src/main/kotlin/com/nexters/bandalart/android/feature/home/ui/StringResource.kt

```diff
@@ -0,0 +1,23 @@
+package com.nexters.bandalart.android.feature.home.ui
+
+import android.content.Context
+import androidx.annotation.StringRes
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+
+sealed class StringResource {
+
+  data class ViewModelStringViewModel(val value: String) : StringResource()
+  class StringResourceText(@StringRes val resId: Int) : StringResource()
+
+  @Composable
+  fun asString() = when (this) {
+    is StringResourceText -> stringResource(resId)
+    is ViewModelStringViewModel -> value
```

Comment:
흠 네이밍이

Reply:
```suggestion
sealed class UiText {

  data class DirectString(val value: String) :UiText()
  class StringResource(@StringRes val resId: Int) : UiText()

  @Composable
  fun asString() = when (this) {
    is DirectString -> value
    is StringResource -> stringResource(resId)
  }

  fun asString(context: Context) = when (this) {
    is DirectString -> value
    is StringResource-> context.getString(resId)
  }
}
```
필립 센세가 명명한 DynamicString 이라는 네이밍이 별로라고 생각해서 바꾼것같은데(나도 이 네이밍이 별로라고 생각하고) 근데 또 달리 다른 네이밍을 생각하자니 애매하긴하네.. 왜 dynamic 이라고 지었는지도 알것같고 
난 resource를 통해 가져오는 것이 아닌 바로 String 사용한다는 의미에서 Direct 라고 제안을 해보긴하는데 이것도 딱 맘에 들진않구만

Reply:
`  @Composable
  fun asString() = when (this) {
    is DirectString -> value
    is StringResource -> stringResource(resId)
  }`

얘는 screen단에서 쓸 줄 알았는데 안 쓰게 되서 지울것 같아. 
네이밍이 문제네

Reply:
받아들이기로 했어..😞